### PR TITLE
feat:follow-unfollow

### PR DESCRIPTION
**Pull request by tushargoyal812**
https://user-images.githubusercontent.com/69681896/168388493-66a3729d-bf61-4c45-b9ec-487603eab710.mp4

### CODE REVIEW

File: my-tweet/src/context/post-context.js [hunk 1]
```diff
@@ -8,8 +8,10 @@ const PostProvider=({children})=>{
     const [newPostData,setNewPostData]=useState()
     const [login,setLogin]=useState({username:"",password:""})
```
Comment (deepak29-git) [inline my-tweet/src/context/post-context.js:9]: If you use more than three ` useState ` then you should use useReducer hook so your code looks more cleaner and readable

File: my-tweet/src/util-functions/follow-handler.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import axios from "axios";
+
+export const followHandler=async(id,setUserFollow)=>{
+    const token=localStorage.getItem('user')
+    try {
+        const response=await axios.post(`/api/users/follow/${id}`,{},{
+            headers: {
+                authorization: token,
```
Comment (deepak29-git) [inline my-tweet/src/util-functions/follow-handler.js:8]: suggestion: you can make function for token and reuse that function wherever you want to use.

File: my-tweet/src/components/right-sidebar.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { usePost } from "../context/post-context"
+import { followHandler } from "../util-functions/follow-handler"
+import { unfollowHandler } from "../util-functions/unfollow-handler"
+import { UnfollowModal } from "./unfollow-modal"
+
+export const RightSidebar=()=>{
+    const {allUsers,setAllUsers,userFollow,setUserFollow,userUnfollow,setUserUnfollow}=usePost()
+    const { isOpen, onOpen, onClose } = useDisclosure()
+
+    const getAllUser=async()=>{
```
Comment (deepak29-git) [inline my-tweet/src/components/right-sidebar.js:13]: suggestion: you can make separate function for api call so that your code looks more clean

**Review by deepak29-git — APPROVED** [review]

**Review by sharmayatendra — APPROVED** [review]
code looks clean
design is consistent
you can look upon suggestion given by another person

File: my-tweet/src/context/post-context.js (hunk 1 above)
Comment (tushargoyal812) [reply my-tweet/src/context/post-context.js:9]: ok will do it

File: my-tweet/src/util-functions/follow-handler.js (hunk 2 above)
Comment (tushargoyal812) [reply my-tweet/src/util-functions/follow-handler.js:8]: ok

File: my-tweet/src/components/right-sidebar.js (hunk 3 above)
Comment (tushargoyal812) [reply my-tweet/src/components/right-sidebar.js:13]: ok will do it

**Pull request merged**